templates/master/00-master/on-prem/files/haproxy-haproxy.yaml

```diff
@@ -23,6 +23,7 @@ contents:
       bind :::9444 v4v6
       mode http
       monitor-uri /haproxy_ready
+      monitor fail if { nbsrv(masters) eq 0 }
```

Comment:
This looks like exactly what we want here. Have you managed to test it?

Reply:
To be clear, we want this to succeed as long as at least 1 master is configured *regardless of whether any of them are healthy*. It should only fail if there are no masters configured. It would be good to test specifically that it will succeed if there are 3 unhealthy masters.

Reply:
I'm fine with this, but I'm pretty sure it doesn't do what you want and won't solve the problem you're hitting. My reading of "the expression uses the nbsrv fetch method to check the number of servers that are up in the webservers backend." says that if all three servers are marked down in the backend then this will fail. Which, I would argue, is perfectly reasonable behavior, but doesn't solve the flapping problem when the backends are unstable.